wrapper/src/main/java/software/amazon/jdbc/plugin/customendpoint/CustomEndpointMonitorImpl.java

```diff
@@ -0,0 +1,229 @@
+    if (!Region.regions().contains(region)) {
+      throw new RuntimeException(
+          Messages.get(
+              "AwsSdk.unsupportedRegion",
+              new Object[] {regionString}));
+    }
+
+    return region;
+  }
+
+  @Override
+  public void registerPluginService(PluginService pluginService) {
+    this.pluginServicesLock.writeLock().lock();
+    try {
+      this.pluginServices.add(pluginService);
+    } finally {
+      this.pluginServicesLock.writeLock().unlock();
+    }
+  }
+
+  @Override
+  public void deregisterPluginService(PluginService pluginService) {
+    this.pluginServicesLock.writeLock().lock();
+    try {
+      this.pluginServices.remove(pluginService);
+    } finally {
+      this.pluginServicesLock.writeLock().unlock();
+    }
+  }
+
+  @Override
+  public boolean shouldDispose() {
+    this.pluginServicesLock.readLock().lock();
+    try {
+      return this.pluginServices.isEmpty();
+    } finally {
+      this.pluginServicesLock.readLock().unlock();
+    }
+  }
+
+  @Override
+  public void close() {
+    this.stop.set(true);
+
+    try {
+      // TODO: the termination timeout is taken from failover2, should it be shorter since it blocks the current thread?
+      if (!this.monitorExecutor.awaitTermination(30, TimeUnit.SECONDS)) {
+        this.monitorExecutor.shutdownNow();
+        this.rdsClient.close();
+      }
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      this.monitorExecutor.shutdownNow();
+      this.rdsClient.close();
+    }
+  }
+
+  @Override
+  public void run() {
+    try {
+      while (!this.stop.get() && !Thread.currentThread().isInterrupted()) {
+        long start = System.nanoTime();
+
+        final Filter customEndpointFilter =
+            Filter.builder().name("db-cluster-endpoint-type").values("custom").build();
+        final DescribeDbClusterEndpointsResponse endpointsResponse =
+            this.rdsClient.describeDBClusterEndpoints(
+                (builder) ->
+                    builder.dbClusterEndpointIdentifier(this.endpointIdentifier).filters(customEndpointFilter));
+        List<DBClusterEndpoint> endpoints = endpointsResponse.dbClusterEndpoints();
+        if (endpoints.size() != 1) {
+          List<String> endpointURLs = endpoints.stream().map(DBClusterEndpoint::endpoint).collect(Collectors.toList());
+          LOGGER.warning(
+              Messages.get("CustomEndpointMonitorImpl.unexpectedNumberOfEndpoints",
+                  new Object[] {
+                      this.endpointIdentifier,
+                      this.region.id(),
+                      endpoints.size(),
+                      endpointURLs
+                  }
+              ));
+
+          // TODO: should we throw an exception and then return? Or wait and then try again?
+          long elapsedTime = System.nanoTime() - start;
+          long sleepDuration = Math.min(0, this.refreshRateNano - elapsedTime);
```

Comment:
I don't think we need such a precision. We can just sleep for specified time despite how much time API call takes.

Reply:
Sure. Do you want me to remove these lines below as well?